src/ExpressionEvaluator/Core/Source/ExpressionCompiler/MetadataUtilities.cs

```diff
@@ -95,8 +96,14 @@ internal static class MetadataUtilities
                 }
                 if (identitiesBuilder != null)
                 {
-                    var identity = metadata.MetadataReader.ReadAssemblyIdentityOrThrow();
+                    var reader = metadata.MetadataReader;
+                    var identity = reader.ReadAssemblyIdentityOrThrow();
                     identitiesBuilder.Add(identity);
+                    // If this assembly has no references, assume it is mscorlib.
+                    if (reader.AssemblyReferences.Count == 0)
+                    {
```

Comment:
The compiler also checks if the assembly defines System.Object, we should do that here too.
See PEAssembly.DeclaresTheObjectClass